packages/react/src/components/primitive/allowed-content.tsx

```diff
@@ -0,0 +1,46 @@
+"use client";
+
+import type { ElementType, Ref } from "react";
+
+import { allowedContentTextLabelGenerator } from "@uploadthing/shared";
+
+import { forwardRefWithAs } from "../../utils/forwardRefWithAs";
+import { PrimitiveSlot, usePrimitiveValues } from "./root";
+import type { HasDisplayName, PrimitiveComponentProps, RefProp } from "./root";
+
+const DEFAULT_ALLOWED_CONTENT_TAG = "div" as const;
+
+export type PrimitiveAllowedContentProps<
+  TTag extends ElementType = typeof DEFAULT_ALLOWED_CONTENT_TAG,
+> = PrimitiveComponentProps<TTag>;
+
+export function AllowedContentFn<
+  TTag extends ElementType = typeof DEFAULT_ALLOWED_CONTENT_TAG,
+>(
+  { children, as, ...props }: PrimitiveAllowedContentProps<TTag>,
+  ref: Ref<HTMLDivElement>,
+) {
+  const { routeConfig, state } = usePrimitiveValues("AllowedContent");
+
+  const Comp = as ?? DEFAULT_ALLOWED_CONTENT_TAG;
+
+  return (
+    <Comp data-state={state} {...props} ref={ref}>
+      <PrimitiveSlot
+        default={allowedContentTextLabelGenerator(routeConfig)}
+        children={children}
+      />
```

Comment:
for slots, I prefer to manually specify children because it looked like its a nested element otherwise, when it actually isn't, but I'm happy to update it